init.lua

```diff
@@ -489,63 +454,63 @@ end
 
 local function get_item_usages(item)
 	local usages = {}
-	for name, def in pairs(reg_items) do
-		if not (def.groups.not_in_craft_guide == 1 or
-				def.groups.not_in_creative_inventory == 1) and
-				get_recipe(name).items and
-				def.description and def.description ~= "" then
-
-			local recipes = get_recipes(name)
-			for i = 1, #recipes do
-				local recipe = recipes[i]
-				if item_in_recipe(item, recipe) then
-					usages[#usages + 1] = recipe
-				end
+
+	for _, recipes in pairs(recipes_cache) do
+		for i = 1, #recipes do
+			local recipe = recipes[i]
+			if item_in_recipe(item, recipe) then
+				usages[#usages + 1] = recipe
 			end
 		end
 	end
 
+	if fuel_cache[item] then
+		usages[#usages + 1] = {type="fuel", width=1, items={item}}
+	end
+
 	return usages
 end
 
 local function get_inv_items(player)
-	local invlist = player:get_inventory():get_list("main")
+	local lists = player:get_inventory():get_lists()
```

Comment:
Is it faster than getting the `main` list directly or there are other reasons to do this?

Reply:
It's so items in the craft grid are included. Thought it should be just `main` and `craft`, not the others.